.mint/python-binary-builder.yml

```diff
@@ -0,0 +1,210 @@
+### required init properties
+### init:
+###  python-version: string
+
+# this intentionally shares an id with Ruby, these generate tons of logs
+concurrency-pools:
+  - id: rwx-research/mint-leaves:binary-builder
+    capacity: 20
+    on-overflow: queue
+
+tasks:
+  - key: install-python
+    timeout-minutes: 20
+    run: |
+      set -ueo pipefail
+
+      # install dependencies
+      sudo apt-get update
+      sudo apt-get install \
+        ca-certificates \
+        dpkg-dev \
+        gcc \
+        gnupg \
+        libbluetooth-dev \
+        libbz2-dev \
+        libc6-dev \
+        libdb-dev \
+        libexpat1-dev \
+        libffi-dev \
+        libgdbm-dev \
+        liblzma-dev \
+        libncursesw5-dev \
+        libreadline-dev \
+        libsqlite3-dev \
+        libssl-dev \
+        make \
+        netbase \
+        tk-dev \
+        tzdata \
+        uuid-dev \
+        wget \
+        xz-utils \
+        zlib1g-dev
+      sudo apt-get clean
+
+      # download python
+      wget -O python.tar.xz "https://www.python.org/ftp/python/${PYTHON_VERSION%%[a-z]*}/Python-$PYTHON_VERSION.tar.xz"
+      wget -O python.tar.xz.asc "https://www.python.org/ftp/python/${PYTHON_VERSION%%[a-z]*}/Python-$PYTHON_VERSION.tar.xz.asc"
+      GNUPGHOME="$(mktemp -d)"
+      export GNUPGHOME
+      gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys \
+        A035C8C19219BA821ECEA86B64E628F8D684696D \
+        E3FF2839C048B25C084DEBE9B26995E310250568 \
+        7169605F62C751356D054A26A821E680E5FA6305 \
+        0D96DF4D4110E5C43FBFB17F2D347EA6AA65421D \
+        97FC712E4C024BBEA48A61ED3A5CA953F73C700D
+      gpg --batch --verify python.tar.xz.asc python.tar.xz
+      gpgconf --kill all
+      rm -rf "$GNUPGHOME" python.tar.xz.asc
+
+      # install python
+      mkdir -p python-src
+      sudo tar --extract --directory python-src --strip-components=1 --file python.tar.xz
+      rm python.tar.xz
+
+      pushd python-src
+      gnuArch="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)"
+      sudo ./configure \
+        --build="$gnuArch" \
+        --enable-loadable-sqlite-extensions \
+        --enable-optimizations \
+        --enable-option-checking=fatal \
+        --enable-shared \
+        --with-lto \
+        --with-system-expat \
+        --without-ensurepip
+
+      nproc="$(nproc)"
+      EXTRA_CFLAGS="$(dpkg-buildflags --get CFLAGS)"
+      LDFLAGS="$(dpkg-buildflags --get LDFLAGS)"
+      LDFLAGS="${LDFLAGS:--Wl},--strip-all"
+      sudo make -j "$nproc" \
+        "EXTRA_CFLAGS=${EXTRA_CFLAGS:-}" \
+        "LDFLAGS=${LDFLAGS:-}" \
+        "PROFILE_TASK=${PROFILE_TASK:-}"
+
+      sudo rm python
+      sudo make -j "$nproc" \
+        "EXTRA_CFLAGS=${EXTRA_CFLAGS:-}" \
+        "LDFLAGS=${LDFLAGS:--Wl},-rpath='\$\$ORIGIN/../lib'" \
+        "PROFILE_TASK=${PROFILE_TASK:-}" \
+        python
```

Comment:
Just curious, are the two makes necessary? Is the second `make ... python` insufficient?

Reply:
I'm not sure. I think this was ripped from a Python dockerfile and I'm terrified to change it 😅

Reply:
I'd like to _think_ they probably do different things

Reply:
https://github.com/docker-library/python/blob/7fe6c4b20ab3173a08299df64987af410bdc8923/3.12/bookworm/Dockerfile#L54-L72